rosidl_generator_py/resource/_msg.py.em

```diff
@@ -114,9 +114,27 @@ class @(spec.base_type.type)(metaclass=Metaclass):
         '_@(field.name)',
 @[end for]@
     ]
+
+    _slot_types = [
+@[for field in spec.fields]@
+@[  if field.type.is_primitive_type() ]@
+@[    if field.type.is_array ]@
+        '@(get_python_type(field.type))[@(field.type.array_size)]',
```

Comment:
This doesn't distinguish between a static array (with a fixed size) and a dynamic array (either with an upper bound or without a bound).

Reply:
True. Do you have a preferred method to denote the different types of arrays? I didn't have a strong intuition here

Reply:
I would suggest either the same notation as in the `.msg` or an object representation. Thinking about it the second one is probably be better in terms of stability with the upcoming change to `.idl` files since the notation will change (see https://github.com/ros2/design/blob/gh-pages/articles/142_idl.md).

Reply:
I know I can use `field.type.array_size.is_upper_bound` for the upper bound arrays but I'm not sure how to check for unbounded arrays. Any hints?

Reply:
Looking at the patch I don't see where the type is being distinguished?

Reply:
My apologies, I got my branches mixed up. All of my changes are in one commit (b9a6a81abc8f75fec9899eb0e9c71b889b448620) but I'm unsure of whether I should cherry pick that commit on top of master or 0.5.2. Since this is targeting master, I assume that I should cherry pick on top of that, but doing so causes the build to fail for unrelated reasons..  

`/home/mike/ws_rqt2/build/rosidl_generator_py/rosidl_generator_py/rosidl_generator_py/msg/_string_arrays_s.c:12:10: fatal error: rosidl_generator_c/primitives_sequence.h: No such file or directory
 #include <rosidl_generator_c/primitives_sequence.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/home/mike/ws_rqt2/build/rosidl_generator_py/rosidl_generator_py/rosidl_generator_py/msg/_nested_s.c:12:10: fatal error: rosidl_generator_c/primitives_sequence.h: No such file or directory
 #include <rosidl_generator_c/primitives_sequence.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/home/mike/ws_rqt2/build/rosidl_generator_py/rosidl_generator_py/rosidl_generator_py/msg/_various_s.c:12:10: fatal error: rosidl_generator_c/primitives_sequence.h: No such file or directory
 #include <rosidl_generator_c/primitives_sequence.h>
`

Is there a .repos available with the most recent ros2 branches that I should use for making PR's to master?

Reply:
@dirk-thomas

Reply:
It depends on what you are trying to build. I would suggest using the `.repos` files from the `master` branch in the `ros2/ros2` repo and building everything from source for your use case. Then a patch based on the master of this repo will work.

(The opposite would be to use the `bouncy` branch of the `.repos` file and then also target the `bouncy` branch in this repo. But usually all contributions are targeting the default branch and only in case of severe bug fixes are being considered for backporting afterwards.)

Reply:
The current state seems to pass on CI: [![Build Status](https://ci.ros2.org/buildStatus/icon?job=ci_linux&build=5548)](https://ci.ros2.org/job/ci_linux/5548/)